pkgs/applications/networking/browsers/palemoon/default.nix

```diff
@@ -112,24 +170,42 @@ in stdenv.mkDerivation rec {
     mk_add_options MOZ_MAKE_FLAGS="-j${if enableParallelBuilding then "$NIX_BUILD_CORES" else "1"}"
     mk_add_options AUTOCONF=${autoconf213}/bin/autoconf
     '
+
+    runHook postConfigure
   '';
 
-  buildPhase = "./mach build";
+  buildPhase = ''
+    runHook preBuild
+
+    ./mach build
+
+    runHook postBuild
+  '';
 
   installPhase = ''
+    runHook preInstall
+
     ./mach install
 
     # Fix missing icon due to wrong WMClass
+    # TODO report upstream
```

Comment:
I looked into this again. In its unpatched state the icon isn't missing on my system now, but Pantheon's dock gets confused by the WM_CLASS value. I dumped some test results on the forum, let's see if it gets fixed / any response: https://forum.palemoon.org/viewtopic.php?f=37&t=26746